packages/xdl/src/Api.ts

```diff
@@ -87,19 +88,22 @@ async function _callMethodAsync(
     };
   }
 
-  if (requestOptions) {
-    if (requestOptions.formData) {
-      let convertedFormData = await _convertFormDataToBuffer(requestOptions.formData);
-      let { data } = convertedFormData;
-      options.headers = {
-        ...options.headers,
-        ...requestOptions.formData.getHeaders(),
-      };
-      options = { ...options, data };
-    } else {
-      options = { ...options, ...requestOptions };
-    }
+  if (!requestOptions.timeout && ConnectionStatus.isOffline()) {
+    options.timeout = 1;
   }
+
+  if (requestOptions.formData) {
+    let convertedFormData = await _convertFormDataToBuffer(requestOptions.formData);
+    let { data } = convertedFormData;
+    options.headers = {
+      ...options.headers,
+      ...requestOptions.formData.getHeaders(),
+    };
+    options = { ...options, data };
+  } else {
+    options = { ...options, ...requestOptions };
```

Comment:
need to be careful here not to clobber our timeout config from above. i think it should be fine but i think ideally we would move the timeout stuff after this